### DISCUSSION TITLE
CDK Graph Plugin Diagram: D2 / PlantUML file formats

### DISCUSSION THREAD

**Discussion by v41e** [aws/aws-pdk#481] [Q&A] [answered]
Hi PDK team,

CDK Graph and its diagram plugin look very useful! Currently the plugin generates raw svg/img.

Will it be possible in the future to generate D2 or PlantUML?

Cheers,

P.S. Why your work for example static website is not part of the cdk lib ?

**Comment by JeremyJonas** [comment, marked as answer]
Hi @valebedu ,

Great to hear you enjoy the Graph Plugin. It is still early days for the tool and we definitely have big visions for it, which definitely includes additional output formats.

Currently it also generated Dot (GraphViz) format, which maybe there are converters available to PlantUML format? So that might be an easier one.

We have considered Viz.js (pretty easy) and draw.io (complex) formats, but D2 should be similar to Viz.js and not to complex - if I remember correctly there are libraries for convert Dot/Svg to Viz and D2.

We are definitely keen for contribution! Please feel free to experiment and create a PR 😄 

Given other priorities, it is unlikely we will have time to develop additional output formats soon, but definitely open to PRs!
